rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistStorageProviderImplTest.java

```diff
@@ -0,0 +1,114 @@
+        LockWhitelistEntry oneOffWhiteListEntry =  addOneOffWhiteListEntry();
+        lockWhitelist.put(getBtcAddress(), oneOffWhiteListEntry);
+
+        whitelistStorageProvider.save(activationConfig);
+
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+        assertEquals(1, actualLockWhitelist.getAll().size());
+    }
+
+    @Test
+    void save_whenIsActiveRSKIP87_ShouldSavedUnlimitedLockWhitelist() {
+        when(activationConfig.isActive(ConsensusRule.RSKIP87)).thenReturn(true);
+        LockWhitelist lockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+        LockWhitelistEntry oneOffWhiteListEntry =  addOneOffWhiteListEntry();
+        LockWhitelistEntry unlimitedWhiteListEntry = addUnlimitedWhiteListEntry();
+        lockWhitelist.put(getBtcAddress(), oneOffWhiteListEntry);
+        lockWhitelist.put(getBtcAddress(), unlimitedWhiteListEntry);
+
+        whitelistStorageProvider.save(activationConfig);
+
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+        assertEquals(2, actualLockWhitelist.getAll().size());
+    }
+
+    @Test
+    void getLockWhitelist_whenLockWhitelistIsNull_shouldReturnZeroEntries() {
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+
+        assertEquals(0, actualLockWhitelist.getAll().size());
+    }
+
+    @Test
+    void getLockWhitelist_whenLockWhitelistIsNotNull_shouldReturnOneEntry() {
+        LockWhitelist lockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+        LockWhitelistEntry oneOffWhiteListEntry =  addOneOffWhiteListEntry();
+        lockWhitelist.put(getBtcAddress(), oneOffWhiteListEntry);
+
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+
+        assertEquals(1, actualLockWhitelist.getAll().size());
+    }
+
+    @Test
+    void getWhitelist_whenIsActiveRSKIP87_shouldReturnTwoEntries() {
+        when(activationConfig.isActive(ConsensusRule.RSKIP87)).thenReturn(true);
+        LockWhitelist lockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+        LockWhitelistEntry oneOffWhiteListEntry =  addOneOffWhiteListEntry();
+        LockWhitelistEntry unlimitedWhiteListEntry = addUnlimitedWhiteListEntry();
+        lockWhitelist.put(getBtcAddress(), oneOffWhiteListEntry);
+        lockWhitelist.put(getBtcAddress(), unlimitedWhiteListEntry);
+
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+
+        assertEquals(2, actualLockWhitelist.getAll().size());
+    }
+
+    private LockWhitelistEntry addOneOffWhiteListEntry() {
```

Comment:
I'm following the same pattern as WhitelistSupportImpl which uses terms such as: `add`, and `remove`. These terms are commonly used as synonyms of `create`, and `delete`.

Reply:
I think those are separate things @wilmerrootstock.

WhitelistSupportImpl uses `add` and `remove` because it's adding and removing from storage.

Here you are creating a `LockWhitelistEntry` object, not adding anything

Reply:
What do you guys think about this method name: `createOneOffWhiteListEntry`? 
Take into account we have 2 methods that return a `LockWhitelistEntry`. One related to `OneOffWhiteListEntry` and another one related to `UnlimitedWhiteListEntry`

`createOneOffWhiteListEntry`
`UnlimitedWhiteListEntry`

Do you agree, @nathanieliov @marcos-iov ?

Reply:
Better now.

Reply:
Done!